### PR TITLE
Issues/745 login button custom event

### PR DESCRIPTION
**Pull request by PetarSimonovic**
-  `DownloadPanel` adds custom events for its Log In and Sign Up buttons to the DOM.
-  The Projects site will have corresponding event listeners that will detect when these events are fired

Part of #745

### CODE REVIEW

File: src/events/WebComponentCustomEvents.js [hunk 1]
```diff
@@ -15,3 +15,9 @@ export const runStartedEvent = webComponentCustomEvent("editor-runStarted");
 
 export const stepChangedEvent = (detail) =>
   webComponentCustomEvent("editor-stepChanged", detail);
+
+export const logInEvent = (detail) =>
+  webComponentCustomEvent("editor-logIn", detail);
+
+export const signUpEvent = (detail) =>
+  webComponentCustomEvent("editor-signUp", detail);
```
Comment (loiswells97) [inline src/events/WebComponentCustomEvents.js:23]: You only need to add the detail if we're passing extra data to the event (see the `codeChangedEvent` for instance), but also if you want to keep it like this for consistency I don't mind...

File: src/components/Menus/Sidebar/DownloadPanel/DownloadPanel.test.js [hunk 2]
```diff
@@ -76,6 +85,22 @@ describe("DownloadPanel", () => {
...
+  });
+
+  test("the sign-up button calls a signUpHandler when clicked", async () => {
+    const signUpButton = screen.getByText(
+      "downloadPanel.signUpButton",
+    ).parentElement;
+    fireEvent.click(signUpButton);
+    await waitFor(() => expect(signUpHandler).toHaveBeenCalled());
+  });
+
```
[the remark this answers is not in the document]
Comment (PetarSimonovic) [reply src/components/Menus/Sidebar/DownloadPanel/DownloadPanel.test.js:103]: Not needed - removed!

File: src/events/WebComponentCustomEvents.js (hunk 1 above)
Comment (PetarSimonovic) [reply src/events/WebComponentCustomEvents.js:23]: Removed!

Comment (loiswells97) [reply src/events/WebComponentCustomEvents.js]: On this, I guess there are two options for consistency here: either make the events with no detail all events or all functions that just return the event... 🤔 Either way I guess we'll need to change how some of them are referenced (either `myCustomEvent` or `myCustomEvent` depending on which we go with)

**Review by loiswells97 — APPROVED** [review]
Looks great 👍

**Pull request merged**